From acf499ab612df8721b84fa22016f2d0020bf9b89 Mon Sep 17 00:00:00 2001
From: Bo-Cun Chen <bc-bocun.chen@mediatek.com>
Date: Tue, 11 Nov 2025 22:04:46 +0800
Subject: [PATCH] arm64: dts: mt7988a: change I2C pinmux for SFP RFB

Without this patch, the kernel cannot access the SFP module through
the I2C bus on the SFP RFB.

Signed-off-by: Bo-Cun Chen <bc-bocun.chen@mediatek.com>
---
 arch/arm64/boot/dts/mediatek/mt7988a-rfb-eth1-sfp.dtso | 2 +-
 arch/arm64/boot/dts/mediatek/mt7988a-rfb-eth2-sfp.dtso | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/mediatek/mt7988a-rfb-eth1-sfp.dtso b/arch/arm64/boot/dts/mediatek/mt7988a-rfb-eth1-sfp.dtso
index ef5e4b3..ba40a11 100644
--- a/arch/arm64/boot/dts/mediatek/mt7988a-rfb-eth1-sfp.dtso
+++ b/arch/arm64/boot/dts/mediatek/mt7988a-rfb-eth1-sfp.dtso
@@ -16,7 +16,7 @@ fragment@0 {
 		target = <&i2c2>;
 		__overlay__ {
 			pinctrl-names = "default";
-			pinctrl-0 = <&i2c2_pins>;
+			pinctrl-0 = <&i2c2_0_pins>;
 			status = "okay";
 		};
 	};
diff --git a/arch/arm64/boot/dts/mediatek/mt7988a-rfb-eth2-sfp.dtso b/arch/arm64/boot/dts/mediatek/mt7988a-rfb-eth2-sfp.dtso
index 83d6532..b9aabd2 100644
--- a/arch/arm64/boot/dts/mediatek/mt7988a-rfb-eth2-sfp.dtso
+++ b/arch/arm64/boot/dts/mediatek/mt7988a-rfb-eth2-sfp.dtso
@@ -16,7 +16,7 @@ fragment@0 {
 		target = <&i2c1>;
 		__overlay__ {
 			pinctrl-names = "default";
-			pinctrl-0 = <&i2c1_pins>;
+			pinctrl-0 = <&i2c1_sfp_pins>;
 			status = "okay";
 		};
 	};
-- 
2.45.2

